### PR TITLE
[t127250] Adding scaled gross weight based on trading units to outbou…

### PR DESCRIPTION
**Pull request by BT-rmartin**
…nd PO lines

Deploy: Nothing to do. Just python changes in frepple

 
<div> Links to Odoo: </div> <ul>
<li><a target="_blank" href="https://odoo.bt-group.com/web#view_type=form&model=project.task&id=127250">[t127250] [mt13136] frePPLe Connector: Include LU Gross Weight</a></li>
</ul>
<div>Affected Modules:</div>
<table><thead><tr><th>Module</th><th>Ext</th></tr></thead>
<tr><td>frepple</td><td>.py, .xsd</td></tr>
</tbody></table>

### CODE REVIEW

File: frepple/models/product_product.py [hunk 1]
```diff
@@ -27,3 +27,6 @@ def _frepple_get_common_fields(self):
         if self.default_code:
             res.append(('stringproperty', 'internalreference', self.default_code))
         return res
+
+    def _get_weight(self):
+        return self.weight
```
Comment (BT-dherreros) [inline frepple/models/product_product.py]: missing new line at the end of the file

File: frepple/scripts/xml/frepple_6.11.0.xsd [hunk 2]
```diff
@@ -1259,18 +1148,17 @@
...
     Frepple also recognizes the following XML processing instructions.
     Note that these instructions are never validated or checked by the parser.
     They are only processed inside frepple.
-
       <?python xxxx ?>
       Include Python code in your model.
   -->
 
-</xsd:schema>
+</xsd:schema>
```
Comment (BT-dherreros) [inline frepple/scripts/xml/frepple_6.11.0.xsd]: missing new line at the end of the file.
This XSD is quite big isn't it?

**Review by BT-dherreros — CHANGES_REQUESTED** [review]

File: frepple/models/product_product.py (hunk 1 above)
Comment (BT-cserra) [inline frepple/models/product_product.py]: **[minor]** ```self.ensure_one()``` to catch calls with more than one record or with no records.

**Review by BT-cserra — APPROVED** [review]
Minor missing line regarding singleton check in method directly using ```self``` instead of the ```for```-loop over the records, that can cause errors in the future (although not as the code is now). Pre-approved.

File: frepple/models/product_product.py (hunk 1 above)
Comment (BT-rmartin) [reply frepple/models/product_product.py]: You are right. Thanks

Comment (BT-rmartin) [reply frepple/models/product_product.py]: You are right. Thanks

File: frepple/scripts/xml/frepple_6.11.0.xsd (hunk 2 above)
Comment (BT-rmartin) [reply frepple/scripts/xml/frepple_6.11.0.xsd]: As provided :) thanks

**Pull request merged**